File: Data Structures & Algorithms/validate-parentheses/submission-3.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        map<char,char> mp = {{'{','}'},{'(',')'},{'[',']'}};

        for(auto x: s){
            if(mp[x]){
                st.push(x);
            }
            else{
                if(st.size() and x==mp[st.top()] ){
                    st.pop();
                }
                else{
                    return false;
                }
            }
        }
        if(st.size()){
            return false;
        }
        return true;
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-3.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for i in tokens:
            if(i.isdigit() or i[1:].isdigit()):
                st.append(int(i));
            
            else:
                v1 = st.pop()
                v2 = st.pop()
                print(v1,i,v2)
                if i == '+':
                    st.append(v2+v1);
                if(i=='-'):
                    st.append(v2-v1);
                if(i=="*"):
                    st.append(v2*v1);
                if(i=="/"):
                    st.append(int(v2/v1));
        return st[0];